bitcoin/src/blockdata/script/tests.rs

```diff
@@ -209,6 +209,13 @@ fn script_builder() {
     );
 }
 
+#[test]
+fn script_builder_with_capacity() {
+    let script = Builder::with_capacity(42);
+
+    assert!(script.into_script().capacity() >= 42);
```

Comment:
wouldn't this be strictly equal to 42?  Nice number choice btw :P

Reply:
This delegates to `ScriptBuf::with_capacity()` which delegates to `Vec::<u8>::with_capacity()` [which says](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.with_capacity) "Constructs a new, empty Vec<T> with **at least** the specified capacity." (emphasis mine). So afaict this *could* allocate more than requested.

Reply:
I had to look this up. Can we add a patch up front that  changes the docs on `ScriptBuf::with_capacity` to be the same as that on `Vec::with_capacity`. Then use the same text in the builder.

```
    /// Constructs a new empty script with at least the specified capacity.
```

Reply:
ref: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.with_capacity

Reply:
I don't remember malloc doing such a thing, although I can see why this would make sense from an allocator optimization standpoint.

Reply:
Yes, this is quite a rabbit hole. I stumbled upon this in the past and at the time the doc was contradicting itself. They resolved it recently. The idea is that some allocators may be able to return more memory because returning less makes no sense, so vec can make use of that memory. `malloc` indeed cannot do that because it doesn't even have the API to report the additional capacity but some other allocators can and Rust supports different allocators. (IDK if jemalloc can do this specifically but it's a popular alternative allocator.)